### ISSUE TITLE
Consider using bash's job control for waiting on parallel graph execution finish

### ISSUE THREAD

**Issue by angelhof** [binpash/pash#353]
By @mgree:

> Well, just running wait will make you wait on all child shells to terminate. bash has job control on by default, so you could try something like:
> ```sh
> while kill -0 %%; do kill %%; wait %%;  done
> ```
> to keep killing the current job until kill -0 says it doesn't exist any more. That won't kill any processes that have been disowned (i.e., removed from the job table).

**Comment by angelhof** [comment]
Doesnt seem relevant to leave open as issue due to its low priority and the fact that the current method is working!
